design-proposals/klip-62-ksqldb-ruby-client.md

```diff
@@ -0,0 +1,53 @@
+# KLIP 62 - ksqlDB Ruby Client
+
+**Author**: Lapo Elisacci (@LapoElisacci) | 
+**Release Target**: TBD | 
+**Status**: _In Discussion_ | 
+**Discussion**: TBD
+
+**tl;dr:** Develop a Ruby HTTP2 Client to request ksqlDB REST API
+
+## Motivation and background
+
+Allow Ruby / Ruby on Rails developers to get started with ksqlDB.
+
+## What is in scope
+
+* A lightweight but complete and configurable Ruby client.
+* Support for both synchronous and asynchronous queries.
+* Documentation and examples
+
+## What is not in scope
+
+* Anything that concerns the ksqlDB development itself.
+* Defining an ORM to dynamically build SQL Statements. (It will get handled by a future project)
+
+## Value/Return
+
+To be able to easily include ksqlDB in any Ruby / Ruby on Rails applications.
+
+## Public APIS
+
+N/A
```

Comment:
we should explicitly point out the APIs for the clients that we want to expose here. For eg: `executeQuery()` for batch results, `streamQuery()` for streaming results etc. The reference for the java client is here https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-clients/java-client/